File: data/readable_format/cpp/file790.cpp
#include <iostream>
#include <vector>

class mo_column {
public:
    static std::vector<std::vector<float>> compute_column(int nz, std::vector<std::vector<float>>& q, std::vector<std::vector<float>>& t, int nproma) {
        std::vector<std::vector<float>> res(nz, std::vector<float>(nproma));
        float c = 5.345f;
        for (int proma = 0; proma < nproma; ++proma) {
            for (int k = 1; k < nz; ++k) {
                t[k][proma] = c * (k + 1);
                q[proma][k] = t[k - 1][proma] + t[k][proma] * c;
            }
            q[proma][nz - 1] *= c;
            for (int i = 0; i < nz; ++i) {
                res[i][proma] = t[i][proma];
            }
        }
        return res;
    }

    static void compute_all(int nz, std::vector<std::vector<float>>& q, std::vector<std::vector<float>>& t, int nproma) {
        auto res = compute_column(nz, q, t, nproma);
    }
};

int main() {
    int nz = 3, nproma = 2;
    std::vector<std::vector<float>> t = {{1.0f, 4.0f}, {2.0f, 5.0f}, {3.0f, 6.0f}};
    std::vector<std::vector<float>> q = {{6.0f, 3.0f}, {5.0f, 2.0f}, {4.0f, 1.0f}};

    mo_column::compute_all(nz, q, t, nproma);

    std::cout << "Testing compute_all:" << std::endl;
    for (int i = 0; i < nz; ++i) {
        for (int j = 0; j < nproma; ++j) {
            std::cout << "t(" << i << "," << j << ") = " << t[i][j] << std::endl;
        }
    }
    for (int i = 0; i < nproma; ++i) {
        for (int j = 0; j < nz; ++j) {
            std::cout << "q(" << i << "," << j << ") = " << q[i][j] << std::endl;
        }
    }

    return 0;
}

File: data/readable_format/fortran/file790.f90
MODULE mo_column
  IMPLICIT NONE
CONTAINS
  FUNCTION compute_column(nz, q, t, nproma) RESULT(res)
    INTEGER, INTENT(IN) :: nz, nproma
    REAL, INTENT(INOUT) :: t(nz, nproma)
    REAL, INTENT(INOUT) :: q(nproma, nz)
    REAL :: res(nz, nproma)
    INTEGER :: k, proma
    REAL :: c = 5.345

    DO proma = 1, nproma
      DO k = 2, nz
        t(k, proma) = c * k
        q(proma, k) = t(k-1, proma) + t(k, proma) * c
      END DO
      q(proma, nz) = q(proma, nz) * c
      res(:, proma) = t(:, proma)
    END DO
  END FUNCTION compute_column

  SUBROUTINE compute_all(nz, q, t, nproma)
    INTEGER, INTENT(IN) :: nz, nproma
    REAL, INTENT(INOUT) :: t(nz, nproma)
    REAL, INTENT(INOUT) :: q(nproma, nz)
    REAL :: res(nz, nproma)
    INTEGER :: k, proma

    res = compute_column(nz, q, t, nproma)
  END SUBROUTINE compute_all
END MODULE mo_column

PROGRAM test_mo_column
  USE mo_column
  IMPLICIT NONE
  INTEGER, PARAMETER :: nz = 3, nproma = 2
  REAL :: t(nz, nproma), q(nproma, nz)
  INTEGER :: i, j

  ! Initialize t and q with some test data
  t = RESHAPE([1.0, 2.0, 3.0, 4.0, 5.0, 6.0], SHAPE(t))
  q = RESHAPE([6.0, 5.0, 4.0, 3.0, 2.0, 1.0], SHAPE(q))

  CALL compute_all(nz, q, t, nproma)

  PRINT *, "Testing compute_all:"
  DO i = 1, nz
     DO j = 1, nproma
        PRINT *, "t(", i, ",", j, ") = ", t(i, j)
        PRINT *, "q(", j, ",", i, ") = ", q(j, i)
     END DO
  END DO
END PROGRAM